ceph/ceph_admin/helper.py

```diff
@@ -200,6 +200,73 @@ def generate_osd_spec(self, spec):
 
         return template.render(spec=spec)
 
+    def generate_mds_spec(self, spec):
+        """
+        Return spec content for mds service
+
+        Note: make sure volume is already created.
+
+        Args:
+            spec: mds service spec config
+
+        spec:
+          - service_type: mds
+            service_id: cephfs
+            unmanaged: boolean    # true or false
+            placement:
+              host_pattern: "*"   # either hosts or host_pattern
+              nodes:
+                - node2
+                - node3
+              label: mds
+        Returns:
+            service_spec
+        """
+        template = self._get_template("mds")
+        node_names = spec["placement"].pop("nodes", None)
+        if node_names:
+            spec["placement"]["hosts"] = []
+            nodes = get_nodes_by_ids(self.cluster, node_names)
+            for node in nodes:
+                spec["placement"]["hosts"].append(node.shortname)
```

Comment:
Would it make sense to have a separate method to return the host list which then can we set to `spec["placement"]["hosts"]`? 

I noticed that we are using these lines across multiple methods.

Reply:
Done.